src/Slider/Slider.jsx

```diff
@@ -521,6 +537,39 @@ const Slider = class Slider extends React.Component {
     return null;
   }
 
+  checkInfinite() {
+    let skipToSlide = null;
+
+    const {
+      carouselStore,
+      currentSlide,
+      infinite,
+      totalSlides,
+    } = this.props;
+
+    if (infinite) {
+      if (currentSlide > totalSlides) {
+        skipToSlide = 1;
+      } else if (currentSlide < 0) {
+        skipToSlide = totalSlides - 1
+      }
+
+      if (skipToSlide) {
```

Comment:
Since you are dealing with indexes, it may be better to do a raw check here so JavaScript doesn't get truthy with you, i.e., `skipToSlide !== null`